### PR TITLE
feat(split-button): added new component

### LINKED ISSUE #1606: split-button: add new component
Issue for ebayui side of split button.
Original issue on skin https://github.com/eBay/skin/issues/1567

### PR DESCRIPTION
**Pull request by agliga**
Closes #1606

__NOTE__ This is required to make it look properly: https://github.com/eBay/skin/issues/1690 Please apply the fixes from that bug when testing. 

## Description
* Added split button new component
* Used snapshot for testing
* Added links to readme to storybook and examples

## References
https://github.com/eBay/ebayui-core/issues/1606

## Screenshots
Screen Shot 2022-03-09 at 12 29 56 PM
Screen Shot 2022-03-09 at 1 33 42 PM

### CODE REVIEW

File: src/components/ebay-split-button/split-button.stories.js [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+            options: ['none', 'loading', 'reset'],
+            control: { type: 'select' },
+            table: {
+                defaultValue: {
+                    summary: 'none',
+                },
+            },
+            type: { category: 'Options' },
+        },
+        a11yButtonText: {
```
Comment (ianmcburnie) [inline src/components/ebay-split-button/split-button.stories.js:68]: Might be better as `a11yButtonLoadingText`?

File: src/components/ebay-split-button/split-button.stories.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    parameters: {
+        docs: {
+            description: {
+                component: readme,
+            },
+        },
+    },
+    argTypes: {
+        renderBody: 'Button',
+        href: {
```
Comment (ianmcburnie) [inline src/components/ebay-split-button/split-button.stories.js:31]: I tried adding `href` to component and it doesn't seem to be honoured.

Comment (agliga) [reply src/components/ebay-split-button/split-button.stories.js:31]: We will need additional styles to get href to work. Skin is missing styles for anchor split button. Will add it to https://github.com/eBay/skin/issues/1690

Comment (ianmcburnie) [reply src/components/ebay-split-button/split-button.stories.js:31]: I see. Thanks!

File: src/components/ebay-split-button/index.marko [hunk 3]
```diff
@@ -28,7 +29,7 @@ static {
         priority=input.priority
         bodyState=input.bodyState
         href=input.href
-        a11y-text=input.a11yButtonText
+        a11y-text=input.a11yLoadingButtonText
```
Comment (ianmcburnie) [inline src/components/ebay-split-button/index.marko:32]: It's `a11yLoadingButtonText` here but `a11yButtonLoadingText` above.

**Pull request merged**